osgar/drivers/lora.py

```diff
@@ -0,0 +1,100 @@
+"""
+  LoRa - Long Range Radio
+
+  This module handles communication with LoRa devices and automatic re-transmission
+  of received messages so that all robots work as re-transmission stations.
+  The messages have limited size 40 bytes.
+
+  At the moment driver supports only transmission of "pose2d" messages so the control
+  center (and all neighbor robots) know about each other.
+
+  Current protocol uses numbers for DeviceID (1, 2, ... 5) for source identification
+  and '|' for data separation. The whole packet is terminated by '\n'. The DeviceID
+  is added in front. The message history is collected in reverse order.
+"""
+from datetime import timedelta
+
+from osgar.node import Node
+from osgar.bus import BusShutdownException
+
+ALIVE_MESSAGE = b'alive'
+
+
+def parse_lora_packet(packet):
+    """
+       Parse block of binary data. The block starts with integer
+       numbers interlaced by '|'. List of transmitter IDs is returned
+       with the actual payload data. In the case of missing transmitted
+       history header None is returned.
+    """
+    s = packet.split(b'|')
+    try:
+        return [int(x) for x in s[:-1]], s[-1].strip()
```

Comment:
So here the variable `s` contains a list of two strings (correct?)
So the first part is actually `int(x) for x in s[0]` (correct?)
So the second part is actually `s[1].strip()` (correct?)
Then the first part is not correct:
```python
>>> s = ['123','data']
>>> [int(x) for x in s[:-1]]
[123]
>>> 
```
versus
```python
>>> s = ['123','data']
>>> [int(x) for x in s[0]]
[1, 2, 3]
>>> 
```
Or I still don't understand what is going on here.

Reply:
See example - it should be b'1|2|3|data'